src/GraphQL/Definitions.hs

```diff
@@ -17,14 +34,33 @@ import qualified GHC.TypeLits (TypeError, ErrorMessage(..))
 import qualified GraphQL.Value as GValue
 import qualified Data.GraphQL.AST as AST
 
--- | Argument operator. Can only be used with Field.
+-- $setup
+-- >>> :set -XDataKinds -XTypeOperators
+
+-- | Argument operator. Can only be used with 'Field'.
+--
+-- Say we have a @@Company@@ object that has a field that shows whether
+-- someone is an employee, e.g.
+--
+-- @
+--   type Company {
+--     hasEmployee(employeeName: String!): String!
+--   }
+-- @
+--
+-- Then we might represent that as:
+--
+-- >>> type Company = Object "Company" '[] '[Argument "employeeName" GValue.String :> Field "hasEmployee" Bool]
```

Comment:
It's not `GValue.String`, it'd be `Text` - the field type is what the handler expects.